### PR TITLE
tests: Revise bootloader-related tests for running in sanitycheck

### PR DESCRIPTION
**Pull request by oyvindronningstad**
- Update tags (add 'b0' to all ++)
- Update whitelist (add nrf5340pdk ++)
- Refactor fw_info test to use S1 building directly
- Make flash_patch a basic ztest instead of just a test.sh, and add a
  counter test-case where flash_patch is turned off.
- Add expected console output lines as "harness_config" in testcase.yaml
  for some tests.

Ref: NCSDK-4566

To test, plug in an nrf53 kit and run the following from the nrf folder:

    ../zephyr/scripts/sanitycheck --testcase-root . --platform nrf5340pdk_nrf5340_cpuapp --device-serial /dev/ttyACM2 --device-testing --west-flash="--erase" -v -N -t b0

This should run 6 tests on-target.

### CODE REVIEW

**Comment by torsteingrindvik** [comment]
```
6 of 6 tests passed (100.00%), 0 failed, 0 skipped with 0 warnings in 41.42 seconds
```
Worked nicely.

File: tests/drivers/flash_patch/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
```
Comment (hakonfam) [inline tests/drivers/flash_patch/CMakeLists.txt:2]:
```suggestion
# Copyright (c) 2020 Nordic Semiconductor ASA
```

Rest of PR (my first comment, I just assume this is done elsewhere as well :D )

File: tests/drivers/flash_patch/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+cmake_minimum_required(VERSION 3.8.2)
```
Comment (hakonfam) [inline tests/drivers/flash_patch/CMakeLists.txt:7]: I think a newer version is used in upstream zephyr, you could check that.

File: tests/drivers/flash_patch/prj_off.conf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+CONFIG_ZTEST=y
```
Comment (hakonfam) [inline tests/drivers/flash_patch/prj_off.conf:7]: Could we use overlays instead of complete `prj.conf` files?

File: west.yml [hunk 4]
```diff
@@ -46,7 +46,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: v2.3.0-rc1-ncs1-rc3
+      revision: pull/322/head
```
Comment (hakonfam) [inline west.yml:49]: Please ensure that this is mentioned in the commit message.

File: tests/drivers/flash_patch/CMakeLists.txt (hunk 2 above)
Comment (oyvindronningstad) [reply tests/drivers/flash_patch/CMakeLists.txt:7]: You're right 3.13.1. Seems NCS samples are still on 3.8.x though. I'll change.

**Comment by oyvindronningstad** [comment]
Removed dependency on sdk-zephyr PR. Instead, run sanitycheck with `--west-flash="--erase"` (updated description).

**Pull request merged**